core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -27,25 +30,40 @@ final case class Uri(
     fragment: Option[Fragment] = None)
     extends QueryOps
     with Renderable {
-  import Uri._
 
-  def withPath(path: Path): Uri = copy(path = path)
+  def withSchemeOption(scheme: Option[Scheme]) = copy(scheme = scheme)
+  def withScheme(scheme: Scheme): Uri = withSchemeOption(Some(scheme))
+  def withoutScheme: Uri = withSchemeOption(None)
 
-  def withFragment(fragment: Fragment): Uri = copy(fragment = Option(fragment))
+  def withPath(path: Path): Uri = copy(path = path)
 
-  def withoutFragment: Uri = copy(fragment = Option.empty[Fragment])
+  def withFragmentOption(fragment: Option[Fragment]) = copy(fragment = fragment)
+  def withFragment(fragment: Fragment): Uri = withFragmentOption(Some(fragment))
+  def withoutFragment: Uri = withFragmentOption(None)
 
   def /(newSegment: Path): Uri = {
-    val encoded = UrlCodingUtils.pathEncode(newSegment)
+    val encoded = pathEncode(newSegment)
     val newPath =
       if (path.isEmpty || path.last != '/') s"$path/$encoded"
       else s"$path$encoded"
     copy(path = newPath)
   }
 
+  def withAuthorityOption(authority: Option[Authority]): Uri = copy(authority = authority)
+  def withAuthority(authority: Authority): Uri = withAuthorityOption(Some(authority))
+  def withoutAuthority: Uri = withAuthorityOption(None)
+
   def host: Option[Host] = authority.map(_.host)
+  def withHostOption(host: Option[Host]) =
```

Comment:
Without looking at the implementation, what does this do...

- when there is an authority
- when there is no authority